compiler/rustc_hir/src/definitions.rs

```diff
@@ -20,22 +20,36 @@ use std::hash::Hash;
 /// Internally the `DefPathTable` holds a tree of `DefKey`s, where each `DefKey`
 /// stores the `DefIndex` of its parent.
 /// There is one `DefPathTable` for each crate.
-#[derive(Clone, Default, Debug)]
+#[derive(Debug)]
 pub struct DefPathTable {
+    stable_crate_id: StableCrateId,
     index_to_key: IndexVec<DefIndex, DefKey>,
-    def_path_hashes: IndexVec<DefIndex, DefPathHash>,
+    // We do only store the local hash, as all the definitions are from the current crate.
+    def_path_hashes: IndexVec<DefIndex, Hash64>,
     def_path_hash_to_index: DefPathHashMap,
 }
 
 impl DefPathTable {
+    fn new(stable_crate_id: StableCrateId) -> DefPathTable {
+        DefPathTable {
+            stable_crate_id,
+            index_to_key: Default::default(),
+            def_path_hashes: Default::default(),
+            def_path_hash_to_index: Default::default(),
+        }
+    }
+
     fn allocate(&mut self, key: DefKey, def_path_hash: DefPathHash) -> DefIndex {
```

Comment:
FWIW it looks like we're also hashing the parent's stable crate ID into the local part of this hash -- https://github.com/rust-lang/rust/blob/5eccfc388e70d45c98b859af5c14f397ae1c206e/compiler/rustc_hir/src/definitions.rs#L120

It's probably pretty cheap (just an extra u64 hashed in) but seems like it *would* be correct to skip that hashing since we're not crossing crate boundaries.

Reply:
This changes a lot of hash-dependent behaviour, so I'll investigate this in a follow-up PR.

Reply:
Perf results don't look particularly meaningful either, so it might not really matter.